apps/docs/content/guides/realtime/authorization.mdx

```diff
@@ -0,0 +1,288 @@
+- The user information sent as part of their [Auth JWT](/docs/guides/auth/jwts)
+- The request headers
+- The Channel topic the user is trying to connect to
+
+<Admonition type="caution">
+
+Increased RLS complexity can impact database performance and connection time, leading to higher connection latency and decreased join rates.
+
+</Admonition>
+
+## Helper functions
+
+You can use the following helper functions when writing RLS policies:
+
+### `realtime.topic`
+
+Returns the Channel topic the user is attempting to connect to.
+
+```sql
+create policy "authenticated users can read all messages in room_1"
+on "realtime"."messages"
+for select
+to authenticated
+using (
+  (select realtime.topic()) = 'room_1'
+);
+```
+
+## Examples
+
+The following examples use this schema:
+
+```sql
+create table public.profiles (
+  email text not null,
+  user_id uuid references auth.users (id)
+);
+
+create table public.rooms_users (
+  topic text references realtime.messages (topic),
+  user_id uuid references auth.users (id),
+  created_at timestamptz default current_timestamp
+);
+
+// Trigger to populate profiles table
+create or replace function insert_user () returns trigger as
+$$
+  begin
+    insert into public.profiles (user_id, email) values (new.id, new.email);
+    return new;
+  end;
+$$ language plpgsql;
+
+create or replace trigger "on_new_auth_create_profile"
+after insert on auth.users for each row
+execute function insert_user ();
+
+revoke execute on function insert_user () from public, anon, authenticated;
+grant execute on function insert_user () TO supabase_auth_admin;
+grant insert on table public.profiles to supabase_auth_admin;
+```
+
+### Broadcast
+
+The `extension` field on the `realtime.messages` table records the message type. For Broadcast messages, the value of `realtime.messages.extension` is `broadcast`. You can check for this in your RLS policies.
+
+#### Allow a user to join (and read) a Broadcast topic
+
+To join a Broadcast Channel, a user must have at least one read or write permission on the Channel topic.
+
+Here, we allow reads (`select`s) for users who are linked to the requested topic within the relationship table `public.room_users`:
+
+```sql
+create policy "authenticated users can join topic"
+on "realtime"."messages"
+for select
+to authenticated
+using (
+exists (
+    select
+      ru.user_id
+    from
+      rooms_users ru
+    where
+      ru.user_id = (select auth.uid())
+      and ru.topic = (select realtime.topic())
+      and realtime.messages.extension in ['broadcast']
+  )
+);
+```
+
+Then, to join a topic with RLS enabled, instantiate the Channel with the `private` option set to `true`.
+
+```typescript
+import { createClient } from 'npm:@supabase/supabase-js@2.38.5'
+const url = 'https://<project_ref>.supabase.com'
+const apikey = '<api_key>'
+
+const client = createClient(url, apikey)
+
+const channel = client.channel('room_1', {
+  config: { private: true },
```

Comment:
If `private` isn't `true`, then does that mean RLS doesn't apply, and anyone can join + read + write to the Channel? I wasn't entirely sure about this, probably worth mentioning up at the topic (under How It Works maybe?)

Reply:
exactly and by default we assume it's false at this point in time

Reply:
Added, wdyt?

Reply:
Perfect!

Reply:
good call out, @charislam.

Let's add a note that mentions if private isn't true then the channel is not going to be checking AuthZ.